pulp_smash/tests/ostree/api_v2/test_sync_publish.py

```diff
@@ -0,0 +1,447 @@
+
+
+class SyncValidFeedTestCase(_BaseTestCase):
+    """With a valid feed given, the sync completes with no reported errors."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Create an OSTree repository with a valid feed and branch."""
+        super(SyncValidFeedTestCase, cls).setUpClass()
+        client = api.Client(cls.cfg, api.json_handler)
+        body = _gen_ostree_repo_body()
+        body['importer_config']['feed'] = _VALID_FEED
+        body['importer_config']['branches'] = [_VALID_BRANCHES[0]]
+        repo = client.post(REPOSITORY_PATH, body)
+        client.response_handler = api.echo_handler
+        cls.sync_report = client.post(
+            urljoin(repo['_href'], 'actions/sync/'),
+            {'override_config': {}}
+        )
+        cls.sync_report.raise_for_status()
+        cls.task_bodies = tuple(
+            utils.poll_spawned_tasks(cls.cfg, cls.sync_report.json()))
+        client.response_handler = api.json_handler
+        cls.resources.add(repo['_href'])
+
+    def test_start_sync_code(self):
+        """Assert the call to sync a repository returns an HTTP 202."""
+        self.assertEqual(self.sync_report.status_code, 202)
+
+    def test_task_error(self):
+        """Assert each task's "error" field is null."""
+        for i, task_body in enumerate(self.task_bodies):
+            with self.subTest(i=i):
+                self.assertIsNone(task_body['error'])
+
+    def test_task_traceback(self):
+        """Assert each task's "traceback" field is null."""
+        for i, task_body in enumerate(self.task_bodies):
+            with self.subTest(i=i):
+                self.assertIsNone(task_body['traceback'])
+
+    def test_task_progress_report(self):
+        """Assert no task's progress report contains error details."""
+        for i, task_body in enumerate(self.task_bodies):
+            for action in task_body['progress_report']['ostree_web_importer']:
+                with self.subTest(i=i):
+                    self.assertEqual(
+                        len(action['error_details']),
+                        0,
+                        task_body
+                    )
+
+
+class SyncWithoutFeedTestCase(_BaseTestCase):
+    """Without provided feed, repo synchronisation fails."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Create repository without feed and run sync on it."""
+        super(SyncWithoutFeedTestCase, cls).setUpClass()
+        body = _gen_ostree_repo_body()
+        client = api.Client(cls.cfg, api.json_handler)
+        repo = client.post(REPOSITORY_PATH, body)
+        client.response_handler = api.echo_handler
+        cls.sync_report = client.post(
+            urljoin(repo['_href'], 'actions/sync/'),
+            {'override_config': {}}
+        )
+        cls.sync_report.raise_for_status()
+        cls.task_bodies = tuple(
+            utils.poll_spawned_tasks(cls.cfg, cls.sync_report.json()))
```

Comment:
Nitpick: please move the closing parentheses down to the same indentation level as the call opening:

```python
cls.task_bodies = tuple(
    utils.poll_spawned_tasks(cls.cfg, cls.sync_report.json())
)
```